pkgs/servers/mattermost/default.nix

```diff
@@ -13,30 +12,23 @@ buildGoPackage rec {
     owner = "mattermost";
     repo = "mattermost-server";
     rev = "v${version}";
-    sha256 = "16yf4p0n3klgh0zw2ikbahj9cy1wcxbwg86pld0yz63cfvfz5ns4";
+    sha256 = "0fyy14hqvsdjh1liyj8i570askc2jqsxcw6jrz9b78mw6qncgg9p";
   };
 
   webApp = fetchurl {
     url = "https://releases.mattermost.com/${version}/mattermost-team-${version}-linux-amd64.tar.gz";
-    sha256 = "0ykp9apsv2514bircgay0xi0jigiai65cnb8q77v1qxjzdyx8s75";
+    sha256 = "139x1rfnv0p37ash5yph0ra7wsvlk7i2rmvgabvszh640ifdw70m";
   };
 
   inherit goPackagePath;
 
-  buildPhase = ''
-    runHook preBuild
-    cd go/src/${goPackagePath}/cmd/platform
-    go install ${buildFlags}
-    runHook postBuild
-  '';
-
-  preInstall = ''
-    mkdir -p $bin
+  postInstall = ''
     tar --strip 1 -C $bin -xf $webApp
+    ln -s $bin/bin/platform $bin/bin/mattermost-platform
   '';
 
-  postInstall = ''
-    ln -s $bin/bin/platform $bin/bin/mattermost-platform
+  postFixup = ''
+    patchelf --set-interpreter $(cat $NIX_CC/nix-support/dynamic-linker) $bin/bin/platform
```

Comment:
Why is this necessary? 

```console
$ /nix/store/6b0jynb0szqj81cxsl8y1ivhswxw3wn9-mattermost-4.9.1-bin/bin/mattermost-platform  --help
```

works for me

```
$ ldd /nix/store/6b0jynb0szqj81cxsl8y1ivhswxw3wn9-mattermost-4.9.1-bin/bin/mattermost-platform
        linux-vdso.so.1 (0x00007f866dfe6000)
        libpthread.so.0 => /nix/store/27x7pinqdsl9f3rpbm8bsszd9fhwq266-glibc-2.27/lib/libpthread.so.0 (0x00007f866dba5000)
        libc.so.6 => /nix/store/27x7pinqdsl9f3rpbm8bsszd9fhwq266-glibc-2.27/lib/libc.so.6 (0x00007f866d7f1000)
        /lib64/ld-linux-x86-64.so.2 => /nix/store/27x7pinqdsl9f3rpbm8bsszd9fhwq266-glibc-2.27/lib64/ld-linux-x86-64.so.2 (0x00007f866ddc4000)
```

Reply:
I'm glad it works for you, but it doesn't for me:

```
$ ldd /nix/store/ph0pc99dkpsiz72lwcz173icylzvphkg-mattermost-4.9.1-bin/bin/mattermost-platform
        linux-vdso.so.1 =>  (0x00007fff81928000)
        libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f6c5f3fd000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f6c5f033000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f6c5f61a000)
```
I assume you are building it on NixOS? I'm building it on Ubuntu 16.04 with nix-build version 1.11.13.